### PR TITLE
Sw ga with d3 charts

### PR DESCRIPTION
**Pull request by steppenwells**
Adds analytics powered by google analytics and displays charts using recharts.

css is probably wonky

### CODE REVIEW

File: public/components/CampaignInformation/Analytics/CampaignDailyTrafficChart.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React, { PropTypes } from 'react'
+import {LineChart, Line, XAxis, YAxis, CartesianGrid, Tooltip, Legend, ResponsiveContainer} from 'recharts'
+
+class CampaignDailyTrafficChart extends React.Component {
+
+  formatDate(millis) {
```
Comment (clloyd) [inline public/components/CampaignInformation/Analytics/CampaignDailyTrafficChart.js]: There's a utility file in master for date conversions, could move there?

File: public/components/CampaignInformation/Analytics/CampaignDailyTrafficChart.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    const date = new Date(millis);
+    return date.toLocaleDateString();
+  }
+
+  tooltipFormatDate(millis) {
+    var date = new Date(millis);
```
Comment (clloyd) [inline public/components/CampaignInformation/Analytics/CampaignDailyTrafficChart.js]: utility file

File: public/components/CampaignInformation/Analytics/CampaignPagesCumulativeTrafficChart.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    return analyticsPalette[i].stroke;
+  }
+  
+  getFillColour(index) {
+    const i = index % analyticsPalette.length;
+    return analyticsPalette[i].fill;
+  }
+
+  formatDate(millis) {
+    const date = new Date(millis);
```
Comment (clloyd) [inline public/components/CampaignInformation/Analytics/CampaignPagesCumulativeTrafficChart.js]: utility file

File: public/components/CampaignInformation/Analytics/CampaignPagesCumulativeTrafficChart.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    return date.toLocaleDateString();
+  }
+
+  tooltipFormatDate(millis) {
+    var date = new Date(millis);
```
Comment (clloyd) [inline public/components/CampaignInformation/Analytics/CampaignPagesCumulativeTrafficChart.js]: utility file

File: public/components/CampaignInformation/Analytics/ContentTrafficChart.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    return analyticsPalette[i].stroke;
+  }
+
+  getFillColour(index) {
+    const i = index % analyticsPalette.length;
+    return analyticsPalette[i].fill;
+  }
+  
+  formatDate(millis) {
+    const date = new Date(millis);
```
Comment (clloyd) [inline public/components/CampaignInformation/Analytics/ContentTrafficChart.js]: utility file 🚀

File: public/components/CampaignInformation/Analytics/ContentTrafficChart.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    return date.toLocaleDateString();
+  }
+
+  tooltipFormatDate(millis) {
+    var date = new Date(millis);
```
Comment (clloyd) [inline public/components/CampaignInformation/Analytics/ContentTrafficChart.js]: utility 📁

**Comment by clloyd** [comment]
Looks good apart from replication of the date logic 👍

File: public/styles/components/campaign-info/analytics/_analytics.scss [hunk 7]
```diff
@@ -0,0 +1,72 @@
+.campaign-info {
+  width: 100%;
+
+}
+
+.analytics-chart__full-width {
+
```
Comment (clloyd) [inline public/styles/components/campaign-info/analytics/_analytics.scss:3]: These should probably have a single pseudo class %analytics-chart

Then the different sizes can be based on that class

```
.analytics-chart--full-width {
 @extend %analytics-chart;
 ...
}
```

**Comment by clloyd** [comment]
Added quick comment about sass

File: public/styles/components/campaign-info/analytics/_analytics.scss (hunk 7 above)
Comment (steppenwells) [reply public/styles/components/campaign-info/analytics/_analytics.scss:3]: This is exactly the kind of help I need with the scss, I know what I'm doing is probably not idiomatic (even though it appears to work).

Also naming classes, never really got my head round all that __ stuff.

**Comment by steppenwells** [comment]
Re date logic - there are a few other duplicated functions (path summary, colour fetching) planned to refactor those as well but wanted to talk about the right place etc.

**Comment by steppenwells** [comment]
@clloyd I've just added the propTypes, please check how I've done it and that this is a sensible way of defining common definitions.

Other than that i think this is done for now.

File: public/propTypes/analytics.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import { PropTypes } from 'react'
+
+export const pageCountStat = PropTypes.shape({
```
Comment (clloyd) [inline public/propTypes/analytics.js]: Just to seperate this from the actual prop.pageCountStat, could we call this pageCountStatPropType or pageCountStatPropDef? I know it's long... but might be simpler when used.

**Comment by clloyd** [comment]
👍 For the proptypes, quick comment about naming.

**Pull request merged**